joomla/joomla-cms
# [4.0] Fix Composer Deprecation Notices

Pull Request for Issue #29794 .

### Summary of Changes
Fixes the composer deprecations using the 3.x code from @SharkyKZ here: #30802 


### Testing Instructions
Test 1: Update a Joomla! 3.10 installation with the 4.0 installation of this PR (see at the end for download link)
Test 2: Run ```composer install``` for this PR
Test 3: Test unit tests


### Actual result BEFORE applying this Pull Request
Result 1: (especially on windows systems) the renamed files from this PR have the incorrect case (or old files are still there)
Result 2: composer does throw deprecation notices



### Expected result AFTER applying this Pull Request
Result 1: (especially on windows systems) the renamed files from this PR have the correct case
Result 2: composer does not throw any deprecation notices
Result 3: unit test are working

Thanks @SharkyKZ for the 3.0 code
Thanks @HLeithner for helping with this PR
